src/Nethermind/Nethermind.JsonRpc/Modules/Eth/SimulateTxExecutor.cs

```diff
@@ -47,6 +47,11 @@ protected override SimulatePayload<TransactionWithSourceDetails> Prepare(Simulat
                             callTransactionModel.Type = TxType.EIP1559;
                         }
 
+                        if (callTransactionModel.BlobVersionedHashes is not null)
+                        {
+                            callTransactionModel.Type = TxType.Blob;
+                        }
+
```

Comment:
Can we encapsulate all code related to tx type to `GetTxType` method?

Reply:
Done